spec/abci++/abci++_basic_concepts_002_draft.md

```diff
@@ -106,95 +156,104 @@ the `LoadSnapshotChunk` method.
 
 ### Other methods
 
 Additionally, there is a [**Flush**](./abci++_methods_002_draft.md#flush) method that is called on every connection,
 and an [**Echo**](./abci++_methods_002_draft.md#echo) method that is just for debugging.
 
 More details on managing state across connections can be found in the section on
-[ABCI Applications](../abci/apps.md).
+[Managing Application State](./abci%2B%2B_app_requirements_002_draft.md#managing-the-application-state-and-related-topics)
 
 ## Next-block execution vs. same-block execution
+
 [&uparrow; Back to Outline](#outline)
 
 In the original ABCI protocol, the only moment when the Application had access to a
-block was after it was decided. This led to a block execution model, called _next-block
-execution_, where some fields hashed in a block header refer to the execution of the
+block was after it was decided. This led to a block execution model, called *next-block
+execution*, where some fields hashed in a block header refer to the execution of the
 previous block, namely:
 
-* the Merkle root of the Application's state
-* the transaction results
-* the consensus parameter updates
-* the validator updates
-
-With ABCI++, an Application may decide to keep using the next-block execution model, by doing all its processing in `FinalizeBlock`;
-however the new methods introduced, `PrepareProposal` and `ProcessProposal` allow
-for a new execution model, called _same-block execution_. An Application implementing
-this execution model, upon receiving a raw proposal via `RequestPrepareProposal`
-and potentially modifying its transaction list,
-fully executes the resulting prepared proposal as though it was the decided block.
-The results of the block execution are used as follows:
-
-* The block execution may generate a set of events. The Application should store these events and return them back to Tendermint during the `FinalizeBlock` call if the block is finally decided.
-* The Merkle root resulting from executing the prepared proposal is provided in
+- the Merkle root of the Application's state
+- the transaction results
+- the consensus parameter updates
+- the validator updates
+
+With ABCI++, an Application may decide to keep using the next-block execution model, by doing all
+its processing in `FinalizeBlock`; however the new methods introduced, `PrepareProposal` and
+`ProcessProposal` disclose the entire proposed block to the Application, allowing for its immediate
+exectution. An Application implementing immediate exectution may additionally wish to store
+certain data resulting from the block's execution in the same block that has just been executed.
+This brings about a new execution model, called *same-block execution*. An Application
+implementing this execution model, upon receiving a raw proposal via `RequestPrepareProposal`
+and potentially modifying its transaction list, fully executes the resulting prepared proposal
+as though it was the decided block (immediate execution), and
```

Comment:
Are we still planning to cut out the same-block execution text? Or is that only for the v0.36.x branch?

Reply:
Yes, but:

* only on v0.36.x branch
* once this one has landed both on master and v0.36.x

Some of the changes in this PR are subtly preparing for that.